core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -202,6 +204,27 @@ public long getDataTimestamp() {
         }
     }
 
+    /**
+     * Opens a connection to the given URL
+     * @param src the url to connect to
+     * @return A {@code URLConnection} for the given src URL
+     * @since TODO
+     */
+    public URLConnection connect(URL src) throws IOException {
+        return ProxyConfiguration.open(src);
```

Comment:
This doesn't seem ideal `ProxyConfiguration.open` is deprecated:
https://github.com/jenkinsci/jenkins/blob/7a7293eaf474dd6154f7e59bc6295c61a3a31dbe/core/src/main/java/hudson/ProxyConfiguration.java#L315

You could potentially use this instead:
https://github.com/jenkinsci/jenkins/blob/7a7293eaf474dd6154f7e59bc6295c61a3a31dbe/core/src/main/java/hudson/ProxyConfiguration.java#L426-L432